airflow/contrib/operators/kubernetes_pod_operator.py

```diff
@@ -158,17 +167,17 @@ def execute(self, context):
                 raise AirflowException(
                     'Pod returned a failure: {state}'.format(state=final_state)
                 )
-            if self.xcom_push:
+            if self.do_xcom_push:
                 return result
         except AirflowException as ex:
             raise AirflowException('Pod Launching failed: {error}'.format(error=ex))
 
     def _set_resources(self, resources):
-        inputResource = Resources()
-        if resources:
-            for item in resources.keys():
-                setattr(inputResource, item, resources[item])
-        return inputResource
+        return Resources(**resources) if resources else Resources()
+
+    def _set_name(self, name):
+        validate_key(name, max_length=63)
```

Comment:
So, with all that into consideration I'd suggest the following change:

```suggestion
        validate_key(name, max_length=244)
```

Considering a maximum char limit of `253` (`244` plus the `9` characters mentioned in my first comment).

Thoughts?

Reply:
Good catch! Agreed :) Do you wanna make a PR to master? Preferably with a small Jira ticket so it can be cherry-picked to 1-10?

Reply:
I can give it a try... (never done this before)
Create a JIRA ticket + PR?
Should I base the PR in branch `v1-10-stable`?

Reply:
Please base it on master (v2.0) - new releases for the v1 branches are mostly commits cherry-picked from master onto the 1-10-test and later the 1-10-stable branches

Reply:
Not sure if it's complete enough, it's the first time I contribute (bear with me :see_no_evil:): https://github.com/apache/airflow/pull/8829

Any feedback is welcome!